pkg/devices/cpu_power_meter.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package devices
+
+// EnergyZone represents a measurable energy or power zone/domain exposed by a power meter.
+// An EnergyZone typically represents a logical zone of the hardware unit, e.g. cpu core, cpu package
+// dram, uncore etc.
+// Reference: https://firefox-source-docs.mozilla.org/performance/power_profiling_overview.html
+type EnergyZone interface {
+
+	// Name() returns the zone name
+	Name() string
+
+	// Index() returns the index of the zone
+	Index() int
+
+	// Path() returns the path from which the energy usage value ie being read
+	Path() string
+
+	// Energy() retuns energy consumed by the zone.
+	// The unit of energy is MicroJoules
+	Energy() Joules
+
+	// MaxEnergy returns  the maximum value of energy usage that can be read.
+	// When energy usage reaches this value, the energy value returned by Energy()
+	// will wrap around and start again from zero.
+	// The unit of MaxEnergy is MicroJoules
```

Comment:
Can we return Joules instead?

Reply:
RAPL returns MicroJoules so it is kept same way.

Reply:
I think we should standarize on just Joules, just like Energy(). There isn't a good reason IMHO to return 2 different units for Energy() and MaxEnergy().

Reply:
The units for `Energy()` and `MaxEnergy()` are same, `MicroJoules`.